### DISCUSSION TITLE
What Growatt Inverter are you using?

### DISCUSSION THREAD

**Discussion by wills106** [wills106/homeassistant-solax-modbus#312] [Q&A]
I need a bit of help from the community.

I don't know how the Growatt Inverters serial number system works.

This is important as on other Inverter's such as SolaX & Solis etc this is used to determine the Inverter model.
This is essential so only Single Phase register show on a Single Phase Inverter and Three Phase registers only show on a Three Phase, prevent battery related registers showing on PV only Inverters etc.

There are also two main Generations covered in plugin_growatt, which use a different subset of registers.

If you own a Growatt Inverter could you please test 'plugin_growatt'.
You will get one of possible outcomes:
- Following error in your log "unrecognized inverter type - serial number : {your serial number}"
- Following error in your log "cannot find serial number, even not for other Inverter"

Could you please let me know your Inverter type, Inverter rating and first 4 digits of the serial number. We can then hopefully build from there and fully support your particular Inverter.

At time of writing this, only reading from the Inverter is implemented.

Edit:
The default baud for Growatt is 9600, but it could be set different, check your Inverter LCD
I can't assist with connecting to RS485 other than what's in your manual.
If you are wanting to connect with an RS485 - Ethernet connection instead of a RS485 - USB adaptor please check the [Wiki for a list of adaptors](https://github.com/wills106/homeassistant-solax-modbus/wiki/Compatible-RS485-Adaptors) known to work with Sofar, Solis & Solax Inverters.

**Comment by Habi1989** [comment]
Hi, hope this great Idea will also work on my Growatt SPH 4600


So 1phase growatt SPH 4600 
Rs485 to eth convert er (waveshare (B) 
SERIAL : NPCQB... 
Dual Mppt 
Seplos BMS diy Accu 15,4 kWh (16s)

Please See also my log:


Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: homeassistant.config_entries
Source: custom_components/solax_modbus/plugin_growatt.py:1855
Integration: SolaX Inverter Modbus (documentation, issues)
First occurred: 17:49:33 (3 occurrences)
Last logged: 18:44:45

Error setting up entry Blibb for solax_modbus
Error setting up entry Growatt for solax_modbus
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 383, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/solax_modbus/__init__.py", line 127, in async_setup_entry
    hub = SolaXModbusHub(hass, name, host, port, modbus_addr, interface, serial_port, baudrate, scan_interval, plugin, config)
  File "/config/custom_components/solax_modbus/__init__.py", line 211, in __init__
    self._invertertype = self.plugin.determineInverterType(self, config)
  File "/config/custom_components/solax_modbus/plugin_growatt.py", line 1855, in determineInverterType
    seriesnumber                       = _read_serialnr(hub, 23)
TypeError: _read_serialnr() missing 1 required positional argument: 'swapbytes'

Thanks in advance
BR

Habi

**Reply by wills106** [reply]
I'm going to look into working on Growatt support.
It seems that the Serial Number is random so I can't base detection off that.

Do you know your Firmware Version?
Does it start:

Main - RA
Secondary - ZCBA

**Comment by Habi1989** [comment]
Hi, 

Just had a look:

Property: RA1.0/RAAa94****/ZCBA-00**/A1B9D0**********

* are deleted Numbers , if they are important I could send them via PM to you.
Thanks in advance 

Habi

**Reply by wills106** [reply]
That should be enough for me to make a start. I might need to change the detection method in the future...

**Reply by wills106** [reply]
I have just released [2023.03.2b8](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/2023.03.2b8) Are you able to see if it detects your Inverter and returns entities?

There is no write support yet, it's just set to read from the Inverter until more testing has taken place.
